### DISCUSSION TITLE
Rendition/audio/caption menu not working outside of documentation sandbox

### DISCUSSION THREAD

**Discussion by xrun1** [muxinc/media-chrome#845] [Q&A] [answered]
I'm trying to use the following code:
```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset=utf-8>
        <script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@3/+esm"></script>
        <script type="module" src="https://cdn.jsdelivr.net/npm/hls-video-element@1/+esm"></script>
    </head>
    <body>
        <media-controller>
            <hls-video
                src="https://stream.mux.com/Sc89iWAyNkhJ3P1rQ02nrEdCFTnfT01CZ2KmaEcxXfB008.m3u8"
                slot="media"
                crossorigin
                muted
            ></hls-video>
            <media-rendition-menu hidden anchor="auto"></media-rendition-menu>
            <media-audio-track-menu hidden anchor="auto"></media-audio-track-menu>
            <media-captions-menu hidden anchor="auto"></media-captions-menu>
            <media-control-bar>
                <media-play-button></media-play-button>
                <media-rendition-menu-button></media-rendition-menu-button>
                <media-audio-track-menu-button></media-audio-track-menu-button>
                <media-captions-menu-button></media-captions-menu-button>
            </media-control-bar>
        </media-controller>
    </body>
</html>
```
When pasted in the [HLS documentation page](https://www.media-chrome.org/docs/en/media-elements/hls-video)'s editor, it works fine, but in a local file, the rendition menu only shows "Auto" while the audio and caption buttons do nothing. It's also impossible to see what the docs editor is doing different because the "open sandbox" link returns this:
```json
{"error":"This endpoint requires the params `files` or `parameters` in order to work. See https://codesandbox.io/docs/importing#create-wizard for more details."}
```

**Comment by luwes** [comment]
could you check if the hls-video preload="auto" attribute makes any difference?

here is an example of it working with rendition menu 
https://codesandbox.io/p/sandbox/media-chrome-rendition-menu-zg32wv?file=%2Findex.html

for renditions and audio tracks to work the video metadata has to be loaded. if preload is none, that information is not known yet until the video is loaded by a user action or preload is changed via API.

**Reply by xrun1** [reply]
It doesn't change anything for me; the link you posted also ends in a 404.
Oddly, using the developer tools, I can see the media controller knows about the renditions (`controllerElement.media.videoRenditions` property).
In 2.2.5 things also worked when I used the experimental `media-...-selectmenu` elements.

**Reply by luwes** [reply]
sorry about that, made it public https://codesandbox.io/p/sandbox/media-chrome-rendition-menu-zg32wv?file=%2Findex.html

**Reply by xrun1** [reply]
```html
<script type="module" src="https://cdn.jsdelivr.net/npm/media-chrome@3/+esm"></script>
```
Adding this to the top is enough to break it. I see media-chrome isn't added the same way as mux-video for which there's no internal folder. Where does this media-chrome comes from?

**Reply by luwes** [reply]
hmm, I'm not seeing that. this works ok on my end with the JS link.
https://codesandbox.io/p/sandbox/media-chrome-rendition-menu-forked-fq9wgl?file=%2Findex.html%3A9%2C11

it's from this example if you click the bottom right button "Open Sandbox"
https://www.media-chrome.org/docs/en/examples/rendition-menu

**Reply by xrun1** [reply, marked as answer]
I got it, it only works if you place the media-chrome script tag *after* mux or hls-video-element's tag, which is the opposite of the example [here](https://www.media-chrome.org/docs/en/media-elements/hls-video). It didn't seem to matter before 3.0.0, is this intended? Thank you for your help.

**Reply by luwes** [reply]
thanks for letting us know, that's not intended. we'll have a look at this soon!

**Reply by luwes** [reply]
this should be fixed in the meantime.
